notifications/bridge.go

```diff
@@ -65,37 +65,41 @@ func (b *bridge) ManifestDeleted(repo distribution.Repository, sm *manifest.Sign
 }
 
 func (b *bridge) LayerPushed(repo distribution.Repository, layer distribution.Layer) error {
-	return b.createLayerEventAndWrite(EventActionPush, repo, layer.Digest())
+	return b.createLayerEventAndWrite(EventActionPush, repo, layer)
 }
 
 func (b *bridge) LayerPulled(repo distribution.Repository, layer distribution.Layer) error {
-	return b.createLayerEventAndWrite(EventActionPull, repo, layer.Digest())
+	return b.createLayerEventAndWrite(EventActionPull, repo, layer)
 }
 
 func (b *bridge) LayerDeleted(repo distribution.Repository, layer distribution.Layer) error {
-	return b.createLayerEventAndWrite(EventActionDelete, repo, layer.Digest())
+	return b.createLayerEventAndWrite(EventActionDelete, repo, layer)
 }
 
 func (b *bridge) createManifestEventAndWrite(action string, repo distribution.Repository, sm *manifest.SignedManifest) error {
-	event, err := b.createManifestEvent(action, repo, sm)
+	manifestEvent, err := b.createManifestEvent(action, repo, sm)
 	if err != nil {
 		return err
 	}
 
-	return b.sink.Write(*event)
+	tagEvent := b.createEvent(EventActionTag)
+	tagEvent.Target = manifestEvent.Target
+
+	return b.sink.Write(*manifestEvent, *tagEvent)
```

Comment:
Any reason we are adding an extra "tag" event? I'd prefer not to propagate that lie until we get the tags API fleshed out.